### PR TITLE
fix: pass an function reference instead of a lambda [AR-1663]

### PR DESCRIPTION
**Pull request by trOnk12**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1663" title="AR-1663" target="_blank">BugAR-1663</a>  Scrolling in the conversation is not smooth and images are flickering
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

While typing a message the ConversationScreen were recomposed causing images and other items being recomposed.

### Causes

The lambda was recreated invalidating the items
 
### Solutions

pass a function reference instead of a lambda

#### How to Test

1. Open a conversation with image assets
2. Start typing

The image should no longer flicker.

### Notes

I found not really intuitive issue causing the Image Assets being recomposed inside the ConversationScreen:

If we have something like this :

```
@Composable
fun SomeParentComposable(someString : String, someLambda : () -> Unit, someOtherLambda : () -> Unit){
SomeOtherInnerComposableTakingOnlyString(someString)
SomeInnerComposable({ someLambda() } , { someOtherLambda() } )
}

@Composable
fun SomeInnerComposable(someLambda : () -> Unit, someOtherLambda : () -> Unit){
... something else here
}
```

and imagine that the someString changes,  SomeOtherInnerComposableTakingOnlyString() will get recomposed of course because String changes, but I was expecting SomeInnerComposable() not being recomposed but because everything gets invalidated inside SomeParentComposable, new lambda is being created and since lambda is an anonymous class object, it gets new hashcode and SomeInnerComposable() gets recomposed a fix to would be to do it like this ;

```
@Composable
fun SomeParentComposable(someString : String, someLambda : () -> Unit, someOtherLambda : () -> Unit){
SomeOtherInnerComposableTakingOnlyString(someString)
SomeInnerComposable(someLambda , someOtherLambda )
}

```
that way we are passing a reference

the same goes for calling a ViewModel functions for example this will recompose SomeInnerComposable,

```
@Composable
fun SomeParentComposable(someString : String,viewModel : ViewModel){
SomeOtherInnerComposableTakingOnlyString(someString)
SomeInnerComposable({ viewModel.someMethod() } , {  viewModel.someOtherMethod() } )
}

@Composable
fun SomeInnerComposable(someLambda : () -> Unit, someOtherLambda : () -> Unit){
... something else here
}

```

this will not because we are passing an object reference :

```
@Composable
fun SomeParentComposable(someString : String,viewModel : ViewModel){
SomeOtherInnerComposableTakingOnlyString(someString)
SomeInnerComposable( viewModel::someMethod , {viewModel::someOtherMethod )
}

@Composable
fun SomeInnerComposable(someLambda : () -> Unit, someOtherLambda : () -> Unit){
... something else here
}
```

### Attachments

not-flickering

----
#### PR Post Merge Checklist for internal contributors

 - [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### 

References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-586/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3382478)(3382478)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3382478)(3382478)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-586/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3382998)(3382998)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3382998)(3382998)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt [hunk 1]
```diff
@@ -51,7 +51,7 @@ fun ConversationScreen(conversationViewModel: ConversationViewModel) {
         onSendButtonClicked = conversationViewModel::sendMessage,
         onSendAttachment = conversationViewModel::sendAttachmentMessage,
         onDownloadAsset = conversationViewModel::downloadAsset,
-        onImageFullScreenMode = { conversationViewModel.navigateToGallery(it) },
+        onImageFullScreenMode =  conversationViewModel::navigateToGallery ,
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt:54]:
```suggestion
        onImageFullScreenMode = conversationViewModel::navigateToGallery,
```

We have to remove those extra spaces =)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt [hunk 2]
```diff
@@ -130,11 +130,11 @@ private fun ConversationScreen(
                             onMessageChanged = onMessageChanged,
                             messageText = conversationViewState.messageText,
                             onSendButtonClicked = onSendButtonClicked,
-                            onShowContextMenu = { message -> conversationScreenState.showEditContextMenu(message) },
+                            onShowContextMenu =  conversationScreenState::showEditContextMenu,
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt:133]:
```suggestion
                            onShowContextMenu = conversationScreenState::showEditContextMenu,
```
same here

**Review by yamilmedina — APPROVED** [review]
LGTM! just some really minor formatting issues (extra spaces) but other than that, good catch 👏

**Review by gongracr — APPROVED** [review]
Those code simplifications look good. However I just tested this buggy behavior on a virtual and physical device and it seems it was somehow already fixed?

In any case these changes do not represent anything that can harm imho, so go ahead 🙌

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-586/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3389362)(3389362)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3389362)(3389362)-dev-release.apk)

**Pull request merged**